src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs

```diff
@@ -436,8 +436,10 @@ private InputModelProperty GetOrCreateProperty(Property property)
                 ConstantValue: property.Schema is ConstantSchema constantSchema ? CreateConstant(constantSchema, constantSchema.Extensions?.Format, property.IsNullable) : null,
                 IsRequired: property.IsRequired,
                 IsReadOnly: property.IsReadOnly,
-                IsDiscriminator: property.IsDiscriminator ?? false,
-                FlattenedNames: property.FlattenedNames.ToList());
+                IsDiscriminator: property.IsDiscriminator ?? false)
+            {
```

Comment:
set the `IsFlattened` also. otherwise the property value may not consistent. e.g. FlattenedNames is not empty, but the `IsFlattened` is false.